src/main/kotlin/de/livepoll/api/controller/PollController.kt

```diff
@@ -36,8 +36,7 @@ class PollController(
 
     @ApiOperation(value = "Get poll", tags = ["Poll"])
     @GetMapping("/{id}")
-    fun getPoll(@PathVariable(name = "id") pollId: Long, @AuthenticationPrincipal user: User): PollDtoOut {
-        accountService.checkAuthorizationByPollId(pollId)
+    fun getPoll(@PathVariable(name = "id") pollId: Long): PollDtoOut {
         return pollService.getPoll(pollId)
```

Comment:
This is critical with regards to security. Nobody should have access to this API endpoint except for the user who created the poll.

Of course, participants who participate in a poll - that was opened voluntarily by a user/admin - should see some attributes of that poll, e.g. the poll name in the frontend. But I'm sure there's another way to grant that instead of having to open up the whole endpoint. What about the websocket calling the service directly instead of the controller? Is that feasible?

Reply:
I agree that this is not the best way, the better solution would be to add the name in the object that is send from the websocket but than we would need more changes in the frontend. For the moment and for the presentation i would remove the authentication. We can add the authentication again after the presentation

Reply:
ok, I approve temporarily if we fix this afterwards (in the next few weeks)